students/mattmaeda/session14/mailroom/mailroom/test/test_cli.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+"""
+Unit tests for mailroom cli
+
+pytest --cov=mailroom
+"""
+
+from unittest import mock
+from io import StringIO
+import pytest
+from mailroom import cli
+
+THANK_YOU_OUTPUT = """
+Dear Matt,
+
+Thank you for your donation of $100.00.
+
+"""
+
+BAD_THANK_YOU_OUTPUT = """Invalid amount 'x' entered.
+
+Dear Matt,
+
+Thank you for your donation of $100.00.
+
+"""
+
+BAD_THANK_YOU_OUTPUT2 = """Invalid amount '0.001' entered.
+
+Dear Matt,
+
+Thank you for your donation of $100.00.
+
+"""
+
+TEST_REPORT_OUTPUT = """
```

Comment:
ideally, you wouldn't have the full report to test against -- this way, if you change anything about the report -- number of spaces, names in the test dataset, whatever, you'll need to updated this.

Though you could wait 'till that happens to refactor :-)

Reply:
Also -- making sure the report is correct really should be in your model tests anyway.